### PR TITLE
fix: signUp Second F&B

### PR DESCRIPTION
**Pull request by nadiia-tsytsylina**
## Describe your changes

add next config for Mui-tel-input;
add onNext Prop;
update F&B functionality
add Check Account endpoint to authApi.
update submit logic;
add error messages in cases if user with such phone/email already exists.
Снимок экрана 2023-11-16 174519
Снимок экрана 2023-11-16 174528

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/sign-up-second/SignUpSecond.tsx [hunk 1]
```diff
@@ -27,13 +31,16 @@ import {
...
   role: 'caregiver' | 'seeker';
+  onNext: () => void;
 }
 
-function SignUpSecond({ role }: IProps): JSX.Element {
+function SignUpSecond({ role, onNext }: IProps): JSX.Element {
   const dispatch = useDispatch();
   const { translate } = useLocales();
   const signUpSecondSchema = useSignUpSecondSchema();
   const dateLength = 10;
```
Comment (lyudad) [inline src/components/sign-up-second/SignUpSecond.tsx:41]: будет переинициализация при каждом перерендере

File: src/redux/api/authAPI.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+interface AccountCheckResponse {
+  statusCode?: number;
+  message?: string;
+}
+
+export const authApi = createApi({
+  reducerPath: 'authApi',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:3001' }),
```
Comment (lyudad) [inline src/redux/api/authAPI.ts:56]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.11

**Comment by nadiia-tsytsylina** [comment]
fix:
add useMemo
add min age for date of birth,
 update authApi, 
 add .env

File: src/components/sign-up-second/SignUpSecond.tsx [hunk 3]
```diff
@@ -1,39 +1,53 @@
...
-  NextButton,
-  StyledForm,
-  StyledDatePicker,
-  ErrorMessage,
-  Wrapper,
-  InputWrapper,
-} from './styles';
+
+import { USER_DATE_BIRTH_FORMAT, EMAIL_ERROR, PHONE_ERROR, getMinBirthDate } from './constants';
+import { ErrorMessage, InputWrapper, NextButton, StyledDatePicker, StyledForm } from './styles';
```
Comment (lyudad) [inline src/components/sign-up-second/SignUpSecond.tsx:26]: пройдись по чеклисту

**Comment by lyudad** [comment]
Screenshot 2023-11-20 at 16 58 14
я не согласна с локейшном этого файла, эти константы могут юзатся в других файлах, вынести в рут

**Comment by nadiia-tsytsylina** [comment]
Fix:

move constants to root;
update styles of birth date field;
update store;

Снимок экрана 2023-11-21 130244
Снимок экрана 2023-11-21 130316

**Comment by nadiia-tsytsylina** [comment]
fix: add custom hook

move main logic from component to custom hook

File: src/components/sign-up-second/SignUpSecond.tsx [hunk 4]
```diff
@@ -44,154 +53,179 @@ function SignUpSecond({ role }: IProps): JSX.Element {
...
+              type: 'manual',
+              message: translate('signUpSecondForm.phoneExist'),
+            });
+          }
+        });
+    } catch (error) {
+      throw new Error(error);
     }
-  );
+  };
```
[the remark this answers is not in the document]
Comment (nadiia-tsytsylina) [reply src/components/sign-up-second/SignUpSecond.tsx:63]: создала отдельный файл hooks. всю основную логику перенесла туда

**Comment by nadiia-tsytsylina** [comment]
> Screenshot 2023-11-20 at 16 58 14 я не согласна с локейшном этого файла, эти константы могут юзатся в других файлах, вынести в рут

константы вынесла в рут

File: src/components/sign-up-second/SignUpSecond.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (nadiia-tsytsylina) [reply src/components/sign-up-second/SignUpSecond.tsx]: fix: add custom hook

move main logic from component to custom hook

Code left in component SignUpSecond:

image

File: src/components/sign-up-second/index.tsx [hunk 5]
```diff
@@ -45,146 +40,146 @@ function SignUpSecondForm({ role, onNext }: IProps): JSX.Element {
...
+                dateFormat={USER_DATE_BIRTH_FORMAT}
+              />
+            )}
+          />
+        </FormControl>
+        {errors?.dateOfBirth && (
+          <ErrorMessage variant="caption">{errors.dateOfBirth?.message}</ErrorMessage>
+        )}
+      </InputWrapper>
+      {role === 'caregiver' && (
```
Comment (lyudad) [inline src/components/sign-up-second/index.tsx:161]: 1.1.5 https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798

Comment (nadiia-tsytsylina) [reply src/components/sign-up-second/index.tsx]: fixed. add user roles to root constants

File: src/components/sign-up-second/SignUpSecond.tsx (hunk 1 above)
Comment (nadiia-tsytsylina) [reply src/components/sign-up-second/SignUpSecond.tsx]: fixed it. moved to constants

File: src/redux/api/authAPI.ts (hunk 2 above)
Comment (nadiia-tsytsylina) [reply src/redux/api/authAPI.ts]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**